test/e2e/specs/editor/various/splitting-merging.spec.js

```diff
@@ -0,0 +1,404 @@
+	} );
+
+	test( 'should forward delete from an empty paragraph', async ( {
+		editor,
+		page,
+	} ) => {
+		// Regression test: Bogus nodes in a RichText container can interfere
+		// with isHorizontalEdge detection, preventing forward deletion.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/8731
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Delete' );
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:paragraph -->
+<p>Second</p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should remove empty paragraph block on backspace', async ( {
+		editor,
+		page,
+	} ) => {
+		// Regression Test: In a sole empty paragraph, pressing backspace
+		// should remove the block.
+		//
+		// See: https://github.com/WordPress/gutenberg/pull/8306
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.press( 'Backspace' );
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe( `` );
+	} );
+
+	test( 'should remove at most one paragraph in forward direction', async ( {
+		editor,
+		page,
+	} ) => {
+		// Regression Test: A forward delete on empty RichText previously would
+		// destroy two paragraphs on the dual-action of merge & remove.
+		//
+		// See: https://github.com/WordPress/gutenberg/pull/8735
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.type( 'First' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Second' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Delete' );
+
+		// Check the content.
+		const content = await editor.getEditedPostContent();
+		expect( content ).toBe(
+			`<!-- wp:paragraph -->
+<p>First</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>Second</p>
+<!-- /wp:paragraph -->`
+		);
+	} );
+
+	test( 'should ensure always a default block', async ( {
+		editor,
+		page,
+	} ) => {
+		// Feature: To avoid focus loss, removal of all blocks will result in a
+		// default block insertion at the root. Pressing backspace in a new
+		// paragraph should not effectively allow removal. This is counteracted
+		// with pre-save content processing to save post consisting of only the
+		// unmodified default block as an empty string.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/9626
+		await editor.insertBlock( { name: 'core/paragraph' } );
+		await page.keyboard.press( 'Backspace' );
+
+		// There is a default block and post title:
+		expect(
+			await page.locator( '.block-editor-block-list__block' )
+		).toBeDefined();
```

Comment:
```suggestion
		await expect(
			page.locator( 'role=document[name=/Empty block/i]' )
		).toBeVisible();
		await expect(
			page.locator( 'role=textbox[name="Add title"i]' )
		).toBeVisible();
```

Maybe we could write this as a role selector? What do you think?